### DISCUSSION TITLE
A few questions

### DISCUSSION THREAD

**Discussion by tonyfirgun** [jnsmalm/pixi3d#28] [Q&A]
Hey there, I’m quite new to programming so please bear with me. I am creating a 2.5D game by having 3D scenes made through blender rendered out and then character sprites drawn over them through pixijs. I am wondering if I could accomplish this with pixi3D

1. Can I detect collisions? If so, where should I start?

2. How can I move and rotate the camera through code?

3. Can I create a mesh object box or sphere through pixi3d? In theory, I wanted to create an invisible box that represents where my 2d character would collide in a 3D world.

Thank you

**Comment by jnsmalm** [comment]
Hi @tonyfirgun 

> Can I detect collisions? If so, where should I start?

Pixi3D doesn't contain any collision detection methods but if you want a simple method for checking if two bounding spheres collides you can do something like this:

```javascript
function isIntersecting(a, b) {
    return Vec3.distance(a.position, b.position) < (a.radius + b.radius)
  }

  let a = Mesh3D.createCube()
  a.radius = 5

  let b = Mesh3D.createCube()
  b.radius = 10

  if (isIntersecting(a, b)) {
    console.log("collision!")
  }
```
> How can I move and rotate the camera through code?

You can move and rotate the camera as any other Container3D object:
```javascript
Camera3D.main.x = 10
Camera3D.main.y = 10
Camera3D.main.z = 10

Camera3D.main.rotationQuaternion.setEulerAngles(0,45,0)
```

> Can I create a mesh object box or sphere through pixi3d? In theory, I wanted to create an invisible box that represents where my 2d character would collide in a 3D world.

You can create a box by:
```javascript
let mesh = Mesh3D.createCube()
```

For creating a sphere mesh you can create a sphere in Blender and then save as a gilt file and then import that in Pixi3D. You can check https://github.com/jnsmalm/pixi3d/blob/develop/examples/src/pbr.js for an example how to load an gilt file.

**Reply by tonyfirgun** [reply]
Thank you for your response.

I apologize as I'm confused with the Camera3D container. Would I have to create a new PIXI3D.Camera and add it to a 3D container and then add that to the pixi.container?  What I'm doing doesn't  seem to be working

> let camera = new PIXI3D.Camera(this.renderer)
            this.Camera3D.addChild(camera)
            this.Camera3D.x = 50
            this.Camera3D.y = 50
            this.Camera3D.z = 50
            this.Camera3D.rotationQuaternion.setEulerAngles(0, 90, 0)
            this.stage.addChild(this.Camera3D)

**Reply by jnsmalm** [reply]
You don't need to create a new camera, it already available at Camera3D.main. You can use exactly as described above, you may need to use PIXI3D namespace if you didn't import it already.

I also encourage you to have a look at the provided examples at https://github.com/jnsmalm/pixi3d/tree/develop/examples/src

```
PIXI3D.Camera3D.main.x = 10
PIXI3D.Camera3D.main.y = 10
PIXI3D.Camera3D.main.z = 10

PIXI3D.Camera3D.main.rotationQuaternion.setEulerAngles(0,45,0)
```
